src/VisualStudio/Core/Def/Preview/PreviewService.cs

```diff
@@ -37,16 +37,19 @@ public PreviewDialogService(IThreadingContext threadingContext, SVsServiceProvid
         public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
             => this;
 
-        public Solution PreviewChanges(
+        public async Task<Solution?> PreviewChangesAsync(
             string title,
             string helpString,
             string description,
-            string topLevelName,
+            string? topLevelName,
             Glyph topLevelGlyph,
             Solution newSolution,
             Solution oldSolution,
+            CancellationToken cancellationToken,
             bool showCheckBoxes = true)
```

Comment:
Should we just move this before the cancellationToken parameter? I get this is standard for our convention, but do we have enough callers to be worth keeping this optional?

Reply:
I just added overloads with/without this.